adapters/repos/db/lsmkv/store.go

```diff
@@ -209,6 +209,8 @@ func (s *Store) ListFiles(ctx context.Context, basePath string) ([]string, error
 func (s *Store) runJobOnBuckets(ctx context.Context,
 	jobFunc jobFunc, rollbackFunc rollbackFunc,
 ) ([]interface{}, error) {
+	s.bucketAccessLock.Lock()
+	defer s.bucketAccessLock.Unlock()
```

Comment:
It may introduce performance penalization as it will be preventing access to the buckets from other routines. If that need to be avoided, I'd suggest `s.bucketAccessLock.Unlock()` to be called right before `wg.Wait()` otherwise it may introduce performance penalization. However it won't be protecting access to the bucket as part of the job being executed, only when iterating over the buckets...

Reply:
good call out!, what I am thinking what we need in many places is bucket level lock to avoid contentions 🤔  but that is a separate topic